trace/trace.go

```diff
@@ -254,13 +255,21 @@ func newSpan() *Span {
 }
 
 // AddField adds a key/value pair to this span
+//
+// Errors are treated as a special case for convenience: if `val` is of type
+// `error` then the key is set to the error's message in the span.
 func (s *Span) AddField(key string, val interface{}) {
 	// The call to event's AddField is protected by a lock, but this is not always sufficient
 	// See send for why this lock exists
 	s.eventLock.Lock()
 	defer s.eventLock.Unlock()
 	if s.ev != nil {
-		s.ev.AddField(key, val)
+		err, ok := val.(error)
+		if ok {
```

Comment:
This could be simplified to:
```suggestion
		if err, ok := val.(error); ok {
```